### DISCUSSION TITLE
Change the viewPort?

### DISCUSSION THREAD

**Discussion by oscarthroedsson** [recharts/recharts#3754] [Q&A] [answered]
Hey! 
New to Recharts and trying to implement it on my web-page. 

When I try to implement the BarChart ([https://recharts.org/en-US/examples/SimpleBarChart](url)) the viewport on the right side is a lot bigger then the left side. I would like to get it even on all four sides. Is their a way I can make this happen? 

You can look on the img I have provided. 
![Skärmavbild 2023-09-12 kl  16 01 49](https://github.com/recharts/recharts/assets/74626974/b44376b4-e1af-4288-baa9-8154f4ff8c5e)

Thanks for the understanding that I am new to this and would like all the help and explanation possible.

**Comment by ckifer** [comment]
Hi, welcome!

There is some default margin on the charts so they fit the space that they are in.

You can adjust that margin to whatever fits your needs. Here it is at 0
https://codesandbox.io/s/simple-bar-chart-forked-7jl977?file=/src/App.tsx

This still has some extra space on the left and the bottom (we need room for the axes) but you should be able to adjust those numbers to what looks best to you

**Reply by oscarthroedsson** [reply]
Hey, and thank you for your response. 

If i understand you correctly, you mean that i can even the sides out with the margin? =) 

This is the result, playing with margin. 
`margin={{
            top: 10,
            right: 80,
            left: 0,
            bottom: 0,
          }}`
![Skärmavbild 2023-09-12 kl  19 35 41](https://github.com/recharts/recharts/assets/74626974/7e9ad9df-25b4-4726-8314-ce1164f17632)

And here it is 0 in margin
![Skärmavbild 2023-09-12 kl  19 39 59](https://github.com/recharts/recharts/assets/74626974/8e41c8db-1a52-4a71-a3a1-6ed11e9755d4)

Isn´t it a better way to display the chart in a mobile view? =)

**Reply by ckifer** [reply]
Are you using `ResponsiveContainer`?

https://codesandbox.io/s/simple-bar-chart-forked-7jl977?file=/src/App.tsx

Your chart shouldn't go out of bounds of your container if you do that.

Other than that you can set negative margin (wouldn't recommend, but maybe it can help a little but without ill effect) or you can set the width of your `YAxis` (also don't recommend because then large numbers will get cut off if that width is fixed)

**Reply by ckifer** [reply]
<img width="334" alt="image" src="https://github.com/recharts/recharts/assets/25180830/166c2774-721a-40e8-960f-e709bacbd037">

**Reply by oscarthroedsson** [reply]
What!?!?!, how did you do that? Before you read on, I wanna say thank you for your time and sorry if I´am annoying. 

INFO: The code below is in a component. Except the two dives, I have another, look at the end to see the div containing the components. 

This is my code to the following img. 
This divs are not showing in my code example: 
<div className="flex flex-col items-center w-full">
<div className="xs:w-4/5 p-3 bg-white rounded-md temp mb-3 ">

`return (
    <div className="flex flex-col items-center w-full">
      
      <div className="xs:w-4/5 p-3 bg-white rounded-md temp ">
      
        <ResponsiveContainer width="100%" height={100}>
      
          <BarChart className="barChart" width={500} height={100} data={data}>
      
            <CartesianGrid />
      
            <XAxis dataKey="name" fontSize={11} />
      
            <YAxis fontSize={11} />
      
            <Tooltip />
      
            <Bar dataKey="pv" fill="#8884d8" />
      
          </BarChart>
      
        </ResponsiveContainer>
      
      </div>
    
    </div>
  );`

![Skärmavbild 2023-09-13 kl  07 58 32](https://github.com/recharts/recharts/assets/74626974/da1c719b-cf3a-4f4d-a0e9-4990226ab6ce)

This is on a wider screen, no matter which breakpoint I look. at, it is the same. 
![Skärmavbild 2023-09-13 kl  08 10 49](https://github.com/recharts/recharts/assets/74626974/d257f6f7-5ad6-4d84-a809-4308b1c8f1c4)

Calling my divs **parent 1 and 2** to make it easy. 

-> Took away margin from p2 (marginbottom = mb-3)
This gave a few px better placement. 

It doesnt matter if I change other class rules, it does not affect the chart in any positive or negative way. 

The div containing the components. We are in WeekData with the code above. 
`<div className="md:xsRelative flex items-center">
          <ListBox></ListBox>
          <CommonCrime></CommonCrime>
          <WeekData></WeekData>
</div>`

**Reply by ckifer** [reply, marked as answer]
I guess its hard to do anything without an actual reproduction. I can wrap things in divs and get the same result as above - everything is as tight to the container as possible without cutting things off. On the left and the bottom
https://codesandbox.io/s/simple-bar-chart-forked-7jl977?file=/src/App.tsx

<img width="305" alt="image" src="https://github.com/recharts/recharts/assets/25180830/7e2af40e-14e4-4cd1-86df-cc5180ad6fb3">

Can you try to modify the sandbox to represent the issue better?

**Reply by oscarthroedsson** [reply]
Here is what i do different, if it matter. 

Code I don´t copy in to the component: 
`import React, { PureComponent } from 'react';`

I have this in my app.jsx 
`import { useState, useEffect } from "react";`

and this i have in my main.jsx
`import React from "react";
import ReactDOM from "react-dom/client";
import Router from "./routes/router";`

Do i need this: `import React, { PureComponent } from 'react';` ?

instead of: 
`
export default class Example extends PureComponent {
  static demoUrl = 'https://codesandbox.io/s/simple-bar-chart-tpz8r';

  render() {
  return(
  )}
  `
  i have this: 
  `export default function WeekData() {
  return (
    <div className="flex flex-col justify-center items-center">
      <div className="xs:w-64 infoContainer">
        <ResponsiveContainer width="100%" height="100%">
          <BarChart
            width={500}
            height={300}
            data={data}
            margin={{
              top: 5,
              right: 30,
              left: 20,
              bottom: 5,
            }}
          >
            <CartesianGrid strokeDasharray="3 3" />
            <XAxis dataKey="name" />
            <YAxis />
            <Tooltip />
            <Bar dataKey="pv" fill="#8884d8" />
            <Bar dataKey="uv" fill="#82ca9d" />
          </BarChart>
        </ResponsiveContainer>
      </div>
      <a href="#" className="text-size0-p highlight px-2">
        Undersök mer
      </a>
    </div>
  );
}`

This is my code in sandbox.
https://codesandbox.io/s/simple-bar-chart-forked-svl9m5?file=/src/App.tsx

And i still have the problem. 
![Skärmavbild 2023-09-13 kl  17 43 33](https://github.com/recharts/recharts/assets/74626974/7a9db136-815f-427a-9eaa-a55a776819cc)

**Reply by oscarthroedsson** [reply]
LoL, didnt see you -margin. 
![Skärmavbild 2023-09-13 kl  17 48 20](https://github.com/recharts/recharts/assets/74626974/8e4eaff5-ada8-438f-b234-e1b0093aed09)

It looks okey.. 

Thank you for your time and help.

**Reply by ckifer** [reply]
Sure thing. We'll look at making this better in the future
